compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -408,6 +421,10 @@ private void visitType(Location loc, BType expType, BType actualType, SymbolEnv
                     findTypeParamInInvokableType(loc, (BInvokableType) expType, (BInvokableType) actualType, env,
                             resolvedTypes, result);
                 }
+                if (actualType.tag == TypeTags.SEMANTIC_ERROR && expr != null && expr.getKind() == NodeKind.LAMBDA) {
```

Comment:
This can be null, please check 
https://github.com/ballerina-platform/ballerina-lang/pull/35460/files#diff-de9cff80105102f5b314d27f1c08c3a65f5034cc0b27e92999d9130bacb6d51cR292